src/neo/Trie/MPT/MPTTrie.cs

```diff
@@ -0,0 +1,374 @@
+using System.Collections.Generic;
+
+namespace Neo.Trie.MPT
+{
+    public class MPTTrie : ITrie
+    {
+        private MPTDatabase db;
+        private MPTNode root;
+
+        public MPTTrie(MPTDatabase db)
+        {
+            if (db is null)
+                throw new System.Exception();
+            this.db = db;
+            var rbytes = db.GetRoot();
+            if (rbytes is null || rbytes.Length == 0)
+            {
+                this.root = HashNode.EmptyNode();
+            }
+            else
+            {
+                this.root = Resolve(rbytes);
+            }
+        }
+
+        public MPTNode Resolve(byte[] hash)
+        {
+            return db.Node(hash);
+        }
+
+        public bool TryGet(byte[] path, out byte[] value)
+        {
+            path = path.ToNibbles();
+            return tryGet(ref root, path, out value);
+        }
+
+        private bool tryGet(ref MPTNode node, byte[] path, out byte[] value)
```

Comment:
```suggestion
        private bool TryGet(ref MPTNode node, byte[] path, out byte[] value)
```

Reply:
We have `TryGet`. It's a public method.
`tryGet` is a private method.

Reply:
Pascal specification requires all methods to be capitalized. 😂 Not matter whether it's a private or public method.

Reply:
I remember that all functions in Pascal specification must be capitalized, regardless of private or public functions.